js/apps/admin-ui/src/user/UserGroups.tsx

```diff
@@ -208,20 +128,20 @@ export const UserGroups = ({ user }: UserGroupsProps) => {
   };
 
   const addGroups = async (groups: GroupRepresentation[]): Promise<void> => {
-    const newGroups = groups;
-
-    newGroups.forEach(async (group) => {
-      try {
-        await adminClient.users.addToGroup({
-          id: user.id!,
-          groupId: group.id!,
-        });
-        refresh();
-        addAlert(t("addedGroupMembership"), AlertVariant.success);
-      } catch (error) {
-        addError("users:addedGroupMembershipError", error);
-      }
-    });
+    try {
+      await Promise.all(
+        groups.map((group) =>
+          adminClient.users.addToGroup({
+            id: user.id!,
+            groupId: group.id!,
+          })
+        )
+      );
+      refresh();
```

Comment:
Some of these Promises could resolve successfully and mutate the data. Should we perhaps always call `refresh()`?